server/log_rpc_server.go

```diff
@@ -564,18 +568,22 @@ func (t *TrillianLogRPCServer) GetLeavesByHash(ctx context.Context, req *trillia
 		return nil, err
 	}
 
-	root, err := tx.LatestSignedLogRoot(ctx)
+	slr, err := tx.LatestSignedLogRoot(ctx)
 	if err != nil {
 		return nil, err
 	}
+	var root types.LogRootV1
```

Comment:
is `root` referenced?

Reply:
It's just being checked for validity (unpacks ok) before we return it to clients.

Reply:
Perhaps a local call to `VerifySignedLogRoot` would be better?

Reply:
It would also be clearer because the returned argument could be set to underscore `_`

Reply:
I think that's beyond the scope of this PR, it's already a big change.

Reply:
Would a TODO be appropriate?

Reply:
Maybe an issue as it's not a trivial change. The tests don't include signatures for one thing so would all need updating.

Reply:
On more thought does it really make sense for the server to check the signature it just pulled from storage? It's a fair point for clients but there isn't really a boundary here. If we add checks we prevent clients from detecting misoperation.

Reply:
There's no boundary here.  Checking the signature would be similar to the unmarshing check: perform the checks on the server to make sure everything is ok before returning it to the client.